benchmarks/bayesmark/optuna_optimizer.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Union
+
+import numpy as np
+
+from bayesmark.abstract_optimizer import AbstractOptimizer
+from bayesmark.experiment import experiment_main
+import optuna
+from optuna import pruners
+from optuna import samplers
+
+
+_SAMPLERS = {
+    "GridSampler": samplers.GridSampler,
+    "RandomSampler": samplers.RandomSampler,
+    "TPESampler": samplers.TPESampler,
+    "CmaEsSampler": samplers.CmaEsSampler,
+    "NSGAIISampler": samplers.NSGAIISampler,
+}
+
+_PRUNERS = {
+    "NopPruner": pruners.NopPruner,
+    "MedianPruner": pruners.MedianPruner,
+    "PatientPruner": pruners.PatientPruner,
+    "PercentilePruner": pruners.PercentilePruner,
+    "SuccessiveHalvingPruner": pruners.SuccessiveHalvingPruner,
+    "HyperbandPruner": pruners.HyperbandPruner,
+    "ThresholdPruner": pruners.ThresholdPruner,
+}
```

Comment:
Note - Currently all pruners are `nop` since vanilla bayesmark does not support partial fit. Leaving for possible future extensions.